Delete feature

stdavis opened the pull request:
I think that this is pretty close. Just need to get the exact signature for the delete service.

steveoh commented on src/app/project/FeatureDetails.js:
```diff
@@ -146,6 +182,28 @@ define([
+            var that = this;
+            var projectId = router.getProjectId();
+            var onErrorWithDefault = lang.partial(onError, 'Error deleting feature!');
+            xhr.del(config.urls.api + '/project/' + projectId + '/feature/delete/' + that.currentId, {
```
I would remove the delete fragment and allow the verb to dictate what is happening. I think it should be the same signature as the feature details with the query string category

stdavis replied:
Addressed.

steveoh commented:
I think we should add a delete confirmation dialog.

stdavis commented:
Great point. Added with ab8418c.

steveoh commented:
We also need the user token and key for authentication/authorization.

stdavis commented:
Addressed in fb8598f.

steveoh commented:
we should also disable the button to guard against double clickers.

steveoh commented:
do we also need to refresh the layer?

stdavis commented:
> disable the button to guard against double clickers

Addressed in 672f119

> show notification of the inflight request

Addressed in a2d1bc1

> change success http status code to 202

Addressed in 5974897

> possibly refresh the layer after success?

That's what [this line](https://github.com/agrc/wri-web/blob/delete-feature/src/app/project/FeatureDetails.js#L207) should do. Is it not working?
